tests/phpunit/rule_test.php

```diff
@@ -71,8 +71,13 @@ public function test_validate_settings_with_invalid_data() {
      */
     public function test_save_settings() {
         global $DB;
-        $quiz = new stdClass();
-        $quiz->id = 1;
+        $course = $this->getDataGenerator()->create_course();
+        $quiz = $this->getDataGenerator()->create_module('quiz', ['course' => $course->id]);
+
+        // Using a generator will create the quiz_settings record.
```

Comment:
@nhoobin did you forget to clean this up?

Reply:
or at least clean up comment ;) " // Lets remove it to emulate an existing quiz prior to installing the plugin."

Reply:
I've updated the test name to reflect the comments.